stl/inc/xmemory

```diff
@@ -2003,6 +2007,99 @@ typename _Container::size_type _Erase_nodes_if(_Container& _Cont, _Pr _Pred) {
     }
     return _Old_size - _Cont.size();
 }
+
+#if _HAS_CXX20
+template <class _Ty, class _Alloc, class... _Types, enable_if_t<!_Is_specialization_v<_Ty, pair>, int> = 0>
+constexpr auto uses_allocator_construction_args(const _Alloc& _Al, _Types&&... _Args) noexcept {
+    if constexpr (!uses_allocator_v<_Ty, _Alloc>) {
+        static_assert(is_constructible_v<_Ty, _Types...>,
+            "If uses_allocator_v<T, Alloc> does not hold, T must be constructible from Types... ");
+        (void) _Al;
+        return _STD forward_as_tuple(_STD forward<_Types>(_Args)...);
+    } else {
+        if constexpr (is_constructible_v<_Ty, allocator_arg_t, const _Alloc&, _Types...>) {

```

Comment:
Consider "flattening" this chain to `else if constexpr` - I believe it would be slightly more readable. (After the first branch, we already need to remember that `uses_allocator_v<_Ty, _Alloc>` is true for the rest of the function.) However, this is OK to keep as-is if you prefer the current form.

Reply:
Done